### PR TITLE
James 3427 remove store managers

### PR DESCRIPTION
**Pull request by mbaechler**
Check JIRA content for the rationals.

This is the first step of a long series

### CODE REVIEW

File: mailbox/api/src/main/java/org/apache/james/mailbox/MailboxManager.java [hunk 1]
```diff
@@ -74,6 +78,23 @@
 
 public interface MailboxManager extends RequestAware, RightManager, MailboxAnnotationManager, SessionProvider {
 
+    char SQL_WILDCARD_CHAR = '%';
+    EnumSet<MailboxManager.MessageCapabilities> DEFAULT_NO_MESSAGE_CAPABILITIES = EnumSet.noneOf(MailboxManager.MessageCapabilities.class);
+    int MAX_ATTEMPTS = 3;
+    Duration MIN_BACKOFF = Duration.ofMillis(100);
+    RetryBackoffSpec RETRY_BACKOFF_SPEC = Retry.backoff(MAX_ATTEMPTS, MIN_BACKOFF);
+
+    static MailboxQuery.UserBound toSingleUserQuery(MailboxQuery mailboxQuery, MailboxSession mailboxSession) {
```
Comment (chibenwa) [inline mailbox/api/src/main/java/org/apache/james/mailbox/MailboxManager.java:87]: Maybe this can be moved in MailboxQuery ?

File: mailbox/store/src/test/java/org/apache/james/mailbox/store/StoreMailboxManagerTest.java [hunk 2]
```diff
@@ -63,165 +66,136 @@
 
 import reactor.core.publisher.Mono;
 
-class StoreMailboxManagerTest {
+interface MailboxManagerContract {
```
Comment (chibenwa) [inline mailbox/store/src/test/java/org/apache/james/mailbox/store/StoreMailboxManagerTest.java:69]: The switch to an interface could have been split into a separate commit to make review easier.

**Review by chibenwa — COMMENTED** [review]
I globally agree with this work.

What is the test coverage when you run MailboxManagerTest ?

**Comment by chibenwa** [comment]
`[ERROR] Failed to execute goal org.apache.maven.plugins:maven-checkstyle-plugin:3.1.0:check (default) on project apache-james-mailbox-cassandra: You have 2 Checkstyle violations. -> [Help 1]`

**Comment by Arsnael** [comment]
I don't say I disagree but that's quite the big change in terms of architecture no? Shouldn't that deserve an ADR being written first about it?

**Comment by mbaechler** [comment]
> I don't say I disagree but that's quite the big change in terms of architecture no? Shouldn't that deserve an ADR being written first about it?

Did you read the JIRA? Is there anything that you miss in it to understand this PR? I can copy/paste the content into an ADR as part of this PR but I definitely want to show some code instead of discussing architecture out of context. If people agree with this move, the ADR is a good way to keep track of the decision, of course.

File: mailbox/api/src/main/java/org/apache/james/mailbox/MailboxManager.java (hunk 1 above)
Comment (mbaechler) [reply mailbox/api/src/main/java/org/apache/james/mailbox/MailboxManager.java:87]: why?

**Review by mbaechler — COMMENTED** [review]
> I globally agree with this work.
> 
> What is the test coverage when you run MailboxManagerTest ?

I will check while fixing the build for this PR.

File: mailbox/api/src/main/java/org/apache/james/mailbox/MailboxManager.java (hunk 1 above)
Comment (chibenwa) [reply mailbox/api/src/main/java/org/apache/james/mailbox/MailboxManager.java:87]: mailboxQuery.toUserBound(session) sounds nice no?

**Comment by Arsnael** [comment]
> Did you read the JIRA? Is there anything that you miss in it to understand this PR? I can copy/paste the content into an ADR as part of this PR but I definitely want to show some code instead of discussing architecture out of context. If people agree with this move, the ADR is a good way to keep track of the decision, of course.

Yes I read the JIRA, yes I understand what you are trying to do here, thus my previous sentence starting by 'I don't say I disagree'. Usually I would say on such a big change you would write an ADR first to see if people agree to such a change or not. If people do not, all the effort you put in there is going to waste... don't you think? 

TBH you are the first one usually to complain about a missing ADR as well usually on such changes and that we should discuss architectural changes before making them... (or am I wrong?)

But as I said, I don't disagree with this, it seems @chibenwa doesn't too, and nobody else spoke so far so... go for it?

**Comment by mbaechler** [comment]
> > Did you read the JIRA? Is there anything that you miss in it to understand this PR? I can copy/paste the content into an ADR as part of this PR but I definitely want to show some code instead of discussing architecture out of context. If people agree with this move, the ADR is a good way to keep track of the decision, of course.
> 
> Yes I read the JIRA, yes I understand what you are trying to do here, thus my previous sentence starting by 'I don't say I disagree'. Usually I would say on such a big change you would write an ADR first to see if people agree to such a change or not. If people do not, all the effort you put in there is going to waste... don't you think?

I think quite the opposite in this case: talking before showing is very often a waste of time because people are trying to figure out what it would look like.

Both activity is taking time: writing an ADR and writing code. Why the time to write ADR would be more of a time investment than writing code?

> TBH you are the first one usually to complain about a missing ADR as well usually on such changes and that we should discuss architectural changes before making them... (or am I wrong?)

I may have asked for some ADR in the past, yes. But I guess I wrote the interesting part of the why in the JIRA ticket, enough for readers to understand the intention. If people reading this ticket and this PR agree about the strategy, there's nothing preventing us to write an ADR to record the architectural decision.

However, it's a community project and usually a ticket + some code is a very good start for a contribution, isn't it? Don't try to apply Linagora practices to other people, please.

> But as I said, I don't disagree with this, it seems @chibenwa doesn't too, and nobody else spoke so far so... go for it?

I still need to fix it

**Comment by mbaechler** [comment]
That will be hard to fix things without any access to CI. Sad.

**Comment by chibenwa** [comment]
There is a conflict on this pr...

I'd be happy to communicate build results as soon as it is fixed ;-)

Cheers,

Benoit

**Comment by Arsnael** [comment]
Output logs lost...

Will not restart a build as there is a conflict with master and it will crash right away. Please fix this first thanks :)

**Comment by chibenwa** [comment]
Is progress made toward this?

Is this PR still needed?

Cheers,

Benoit

**Comment by chibenwa** [comment]
Many merge conflict, inactive for very long, opened on linagora repo and not the ASF one...

Reopen on ASF repository if relevant.
